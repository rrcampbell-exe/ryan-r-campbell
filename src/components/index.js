import BlogTags from './Blog/BlogTags'
import BlogTeaser from './Blog/BlogTeaser'
import Books from './Books/Books'
import Builds from './Builds/Builds'
import Contact from './Contact/Contact'
import Footer from './Footer/Footer'
import Header from './Header/Header'
import PageWrapper from './PageWrapper/PageWrapper'
import SubNav from './SubNav/SubNav'
import Reviews from './Reviews/Reviews'
import NotFound from './NotFound/NotFound'
import NotificationBanner from './NotificationBanner/NotificationBanner'
import Loading from './Loading/Loading'
import Quote from './Quote/Quote'
import TextAndImage from './TextAndImage/TextAndImage'
import HamburgerMenu from './HamburgerMenu/HamburgerMenu'
import PopText from './PopText/PopText'
import PodcastTeaser from './Podcast/PodcastTeaser'
import Boasts from './Tech/Boasts'
import SkillsCloud from './Tech/SkillsCloud'
import BlogFooter from './Blog/BlogFooter'
import SubstackRedirect from './SubstackRedirect/SubstackRedirect'
import { CodeSpan, CodeBlock } from './CodeBlock/CodeBlock'

export { BlogTeaser, Books, Builds, Footer, Header, PageWrapper, SubNav, Contact, Reviews, NotFound, NotificationBanner, Loading, Quote, TextAndImage, BlogTags, HamburgerMenu, PopText, PodcastTeaser, Boasts, SkillsCloud, BlogFooter, CodeSpan, CodeBlock, SubstackRedirect }
