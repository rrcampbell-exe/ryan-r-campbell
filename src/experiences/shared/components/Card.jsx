import React from 'react'
import styled from 'styled-components'

// TODO: size should flex based on number of cards
// TODO: need to set up theme colors

const StyledDiv = styled.div`
  height: 16rem;
  width: 16rem;
  border: 1px solid #242424;
  border-radius: 4px;
  padding: 1rem;
  margin: 1rem;
`

const Card = ({ children }) => {
  return (
    <StyledDiv>
      {children}
    </StyledDiv>
  )
}

export default Card
