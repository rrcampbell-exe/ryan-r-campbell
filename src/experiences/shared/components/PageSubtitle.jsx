import React from 'react'
import styled from 'styled-components'

const StyledH2 = styled.h2`
  margin: 0 auto;
  padding: 0;
`

const PageSubtitle = ({ subtitle }) => {
  return (
    <StyledH2>
      {subtitle}
    </StyledH2>
  )
}

export default PageSubtitle
