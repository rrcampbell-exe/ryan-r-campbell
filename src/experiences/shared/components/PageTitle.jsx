import React from 'react'
import styled from 'styled-components'

const StyledH1 = styled.h1`
  margin: 0 auto;
  padding: 0;
`

const PageTitle = ({ title }) => {
  return (
    <StyledH1>
      {title}
    </StyledH1>
  )
}

export default PageTitle
