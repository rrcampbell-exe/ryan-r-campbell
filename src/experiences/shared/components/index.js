import BlogPost from './Blog/BlogPost'
import BlogTeaser from './Blog/BlogTeaser'
import Books from './Books/Books'
import Builds from './Builds/Builds'
import Footer from './Footer/Footer'
import Header from './Header/Header'
import PageWrapper from './PageWrapper/PageWrapper'
import SubNav from './SubNav/SubNav'
import Contact from './Contact/Contact'
import Reviews from './Reviews/Reviews'
import NotFound from './NotFound/NotFound'
import NotificationBanner from './NotificationBanner/NotificationBanner'

export { BlogPost, BlogTeaser, Books, Builds, Footer, Header, PageWrapper, SubNav, Contact, Reviews, NotFound, NotificationBanner }
