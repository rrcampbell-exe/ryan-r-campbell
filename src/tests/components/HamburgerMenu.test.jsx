import { describe, expect, test } from 'vitest'
import { HamburgerMenu } from '../../experiences/shared/components'
import { render } from '@testing-library/react'
import ContextAndRouterProvider from '../test-utils'

describe('HamburgerMenu', () => {
  test('default snapshot', () => {
    const { asFragment } = render(
      <ContextAndRouterProvider>
        <HamburgerMenu />
      </ContextAndRouterProvider>
    )
    expect(asFragment()).toMatchSnapshot()
  })
  test('should close the menu when the close button is clicked', () => {
    const { getByTestId, queryByTestId } = render(
      <ContextAndRouterProvider>
        <HamburgerMenu />
      </ContextAndRouterProvider>
    )
    const closeButton = getByTestId('close-button')
    closeButton.click()
    const menu = queryByTestId('menu')
    expect(menu).not.toHaveClass('menu-open')
  })
  test('should have six link components', () => {
    const { getAllByRole } = render(
      <ContextAndRouterProvider>
        <HamburgerMenu />
      </ContextAndRouterProvider>
    )
    const links = getAllByRole('link')
    expect(links.length).toBe(6)
  })
  test('should show the active link', () => {
    const { getAllByRole } = render(
      <ContextAndRouterProvider>
        <HamburgerMenu />
      </ContextAndRouterProvider>
    )
    const links = getAllByRole('link')
    const activeLink = links.find(link => link.classList.contains('active'))
    expect(activeLink).toBeInTheDocument()
  })
})
