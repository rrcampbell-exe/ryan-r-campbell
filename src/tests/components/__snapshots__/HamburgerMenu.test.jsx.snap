// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HamburgerMenu > default snapshot 1`] = `
<DocumentFragment>
  <div
    class="Hamburger-menu "
    data-testid="menu"
  >
    <button
      class="theme-toggle-button"
    >
      <svg
        id="hamburger-icon"
        version="1.1"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <title>
          Menu
        </title>
        <g
          fill="none"
          fill-rule="evenodd"
          id="Page-1"
          stroke="none"
          stroke-width="1"
        >
          <g
            id="Menu"
          >
            <rect
              fill-rule="nonzero"
              height="24"
              id="Rectangle"
              width="24"
              x="0"
              y="0"
            />
            <line
              id="Path"
              stroke-linecap="round"
              stroke-width="1.5"
              x1="5"
              x2="19"
              y1="7"
              y2="7"
            />
            <line
              id="Path"
              stroke-linecap="round"
              stroke-width="1.5"
              x1="5"
              x2="19"
              y1="17"
              y2="17"
            />
            <line
              id="Path"
              stroke-linecap="round"
              stroke-width="1.5"
              x1="5"
              x2="19"
              y1="12"
              y2="12"
            />
          </g>
        </g>
      </svg>
    </button>
    <div
      class="menu-content"
    >
      <div
        class="icon-container"
      >
        <button
          data-testid="close-button"
        >
          <svg
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20.7457 3.32851C20.3552 2.93798 19.722 2.93798 19.3315 3.32851L12.0371 10.6229L4.74275 3.32851C4.35223 2.93798 3.71906 2.93798 3.32854 3.32851C2.93801 3.71903 2.93801 4.3522 3.32854 4.74272L10.6229 12.0371L3.32856 19.3314C2.93803 19.722 2.93803 20.3551 3.32856 20.7457C3.71908 21.1362 4.35225 21.1362 4.74277 20.7457L12.0371 13.4513L19.3315 20.7457C19.722 21.1362 20.3552 21.1362 20.7457 20.7457C21.1362 20.3551 21.1362 19.722 20.7457 19.3315L13.4513 12.0371L20.7457 4.74272C21.1362 4.3522 21.1362 3.71903 20.7457 3.32851Z"
              stroke-width="0.5"
            />
          </svg>
        </button>
      </div>
      <div
        class="link-container"
      >
        <a
          class="hamburger-link active"
          href="/"
        >
          home
        </a>
        <a
          class="hamburger-link"
          href="/about"
        >
          about
        </a>
        <a
          class="hamburger-link"
          href="/blog"
        >
          blog
        </a>
        <a
          class="hamburger-link"
          href="/author"
        >
          author hub
        </a>
        <a
          class="hamburger-link"
          href="/tech"
        >
          tech hub
        </a>
        <a
          class="hamburger-link"
          href="/podcasts"
        >
          podcasts
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
